lumicks/pylake/population/dwelltime.py

```diff
@@ -7,23 +7,32 @@
 from deprecated.sphinx import deprecated
 
 
-@dataclass
+@dataclass(frozen=True)
 class DwelltimeBootstrap:
     """Bootstrap distributions for a dwelltime model.
 
-    This class is stored in the `DwelltimeModel.bootstrap` attribute
-    and should not be constructed manually.
+    This class should be initialized using :meth:`lk.DwelltimeModel.calculate_bootstrap()
+    <lumicks.pylake.DwelltimeModel.calculate_bootstrap>` and should not be constructed manually.
 
     Attributes
     ----------
-    _samples : np.ndarray
-        array of optimized model parameters for each bootstrap sample pull; shape is
-        [number of parameters, number of samples]
+    model : :class:`~lumicks.pylake.DwelltimeModel`
+        Original model sampled for the bootstrap distribution
+    amplitude_distributions : np.ndarray
+        Array of sample optimized amplitude parameters; shape is [number of components, number of samples]
+    lifetime_distributions : np.ndarray
+        Array of sample optimized lifetime parameters; shape is [number of components, number of samples]
     """
 
-    _samples: np.ndarray = field(default_factory=lambda: np.array([]), repr=False)
+    model: "DwelltimeModel"
+    amplitude_distributions: np.ndarray = field(default_factory=lambda: np.array([]), repr=False)
+    lifetime_distributions: np.ndarray = field(default_factory=lambda: np.array([]), repr=False)
 
-    def _sample_distributions(self, optimized, iterations):
+    def __post_init__(self):
+        assert self.amplitude_distributions.shape == self.lifetime_distributions.shape
```

Comment:
Would be nice to check whether data is consistent with the model provided (i.e. check whether the arrays are two dimensional and whether `n_components` of the model provided corresponds to the shape of dimension `1`).